raiden_contracts/tests/test_unit_internals.py

```diff
@@ -0,0 +1,246 @@
+
+    all_invalid = chain(
+        product(VALID_VALUES, INVALID_VALUES),
+        product(INVALID_VALUES, VALID_VALUES),
+    )
+    for a, b in all_invalid:
+        with pytest.raises(ValidationError):
+            token_network_test.functions.maxPublic(a, b).call()
+
+
+def test_min(token_network_test):
+
+    VALUES = [UINT256_MIN, 1, UINT256_MAX, UINT256_MAX]
+    for a, b in product(VALUES, VALUES):
+        assert token_network_test.functions.minPublic(a, b).call() == min(a, b)
+
+
+def test_max(token_network_test):
+
+    VALUES = [UINT256_MIN, 1, UINT256_MAX, UINT256_MAX]
+    for a, b in product(VALUES, VALUES):
+        assert token_network_test.functions.maxPublic(a, b).call() == max(a, b)
+
+
+def test_recoverAddressFromWithdrawMessage(
+        token_network_test,
+        create_withdraw_signatures,
+        create_channel_and_deposit,
+        get_accounts
+):
+
+    (A, B) = get_accounts(2)
+    deposit_A = 5
+    deposit_B = 7
+    withdraw_A = 3
+    channel_identifier = create_channel_and_deposit(A, B, deposit_A, deposit_B)
+    (signature_A, signature_B) = create_withdraw_signatures(
+        [A, B],
+        channel_identifier,
+        A,
+        withdraw_A,
+        token_network_test.address
+    )
+    token_network_test.functions.verifyWithdrawSignaturesPublic(
+        channel_identifier,
+        A,
+        B,
+        3,
+        signature_A,
+        signature_B
+    ).call()
+
+    with pytest.raises(TransactionFailed):
+        token_network_test.functions.verifyWithdrawSignaturesPublic(
+            channel_identifier,
+            A,
+            B,
+            3,
+            signature_B,
+            signature_A
+        ).call()
+    with pytest.raises(TransactionFailed):
+        token_network_test.functions.verifyWithdrawSignaturesPublic(
+            channel_identifier,
+            A,
+            B,
+            3,
+            signature_A,
+            fake_bytes(64)
+        ).call()
+    with pytest.raises(TransactionFailed):
+        token_network_test.functions.verifyWithdrawSignaturesPublic(
+            channel_identifier,
+            A,
+            B,
+            3,
+            fake_bytes(64),
+            signature_B
+        ).call()
+
+    recovered_address_A = token_network_test.functions.recoverAddressFromWithdrawMessagePublic(
+        channel_identifier,
+        A,
+        withdraw_A,
+        signature_A
+    ).call()
+    assert recovered_address_A == A
+
+    recovered_address_B = token_network_test.functions.recoverAddressFromWithdrawMessagePublic(
+        channel_identifier,
+        A,
+        withdraw_A,
+        signature_B
+    ).call()
+    assert recovered_address_B == B
+
+    with pytest.raises(TransactionFailed):
+        token_network_test.functions.recoverAddressFromWithdrawMessagePublic(
+            channel_identifier,
+            A,
+            withdraw_A,
+            fake_hex(64)
+        ).call()
+
+    assert recovered_address_A != \
```

Comment:
Code style: instead of `\` , please use an additional variable to store the contract call result.